anomalib/deploy/inferencers/torch.py

```diff
@@ -0,0 +1,162 @@
+"""This module contains Torch inference implementations."""
+
+# Copyright (C) 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from pathlib import Path
+from typing import Dict, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from omegaconf import DictConfig, ListConfig
+from torch import Tensor, nn
+
+from anomalib.core.model import AnomalyModule
+from anomalib.data.transforms.pre_process import PreProcessor
+from anomalib.models import get_model
+
+from .base import Inferencer
+
+
+class TorchInferencer(Inferencer):
+    """PyTorch implementation for the inference.
+
+    Args:
+        config (DictConfig): Configurable parameters that are used
+            during the training stage.
+        model_source (Union[str, Path, AnomalyModule]): Path to the model ckpt file or the Anomaly model.
+        meta_data_path (Union[str, Path], optional): Path to metadata file. If none, it tries to load the params
+                from the model state_dict. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        config: Union[DictConfig, ListConfig],
+        model_source: Union[str, Path, AnomalyModule],
+        meta_data_path: Union[str, Path] = None,
+    ):
+        self.config = config
+        if isinstance(model_source, AnomalyModule):
+            self.model = model_source
+        else:
+            self.model = self.load_model(model_source)
+
+        self.meta_data = self._load_meta_data(meta_data_path)
+
+    def _load_meta_data(self, path: Optional[Union[str, Path]] = None) -> Union[Dict, DictConfig]:
+        """Load metadata from file or from model state dict.
+
+        Args:
+            path (Optional[Union[str, Path]], optional): Path to metadata file. If none, it tries to load the params
+                from the model state_dict. Defaults to None.
+
+        Returns:
+            Dict: Dictionary containing the meta_data.
+        """
+        meta_data: Union[DictConfig, Dict[str, Union[float, np.ndarray, Tensor]]] = {}
+        model_params = self.model.state_dict()
+        meta_data_params = {
```

Comment:
The model is already loaded here, so we can just get the metadata from the model itself:
```
image_threshold = self.model.image_threshold.value
pixel_threshold = self.model.pixel_threshold.value
image_mean = self.model.training_distribution.image_mean
...
```